synapse/server_notices/server_notices_manager.py

```diff
@@ -160,6 +160,27 @@ async def get_or_create_notice_room_for_user(self, user_id: str) -> str:
                 self._config.servernotices.server_notices_mxid_display_name,
                 self._config.servernotices.server_notices_mxid_avatar_url,
             )
+            await self._update_room_info(
+                requester,
+                room_id,
+                EventTypes.Name,
+                "name",
+                self._config.servernotices.server_notices_room_name,
+            )
+            await self._update_room_info(
+                requester,
+                room_id,
+                EventTypes.RoomAvatar,
+                "url",
+                self._config.servernotices.server_notices_room_avatar_url,
+            )
+            await self._update_room_info(
+                requester,
+                room_id,
+                EventTypes.Topic,
+                "topic",
+                self._config.servernotices.server_notices_room_topic,
+            )
```

Comment:
I'm not sure it is an issue, but this means that when sending a server notice we'll now fetch 3 more state events and potentially send 3 more events per-user. That's rather heavy, but I don't see a better way to easily batch this.

Reply:
`get_or_create_notice_room_for_user` is cached however, so it should happen only once per restart.

Reply:
Good shout. 👍